libs/oracledb/langchain_oracledb/vectorstores/oraclevs.py

```diff
@@ -0,0 +1,1929 @@
+            await cursor.execute(f"select 1 from {table_name} where ROWNUM < 1")
+            return True
+    except oracledb.DatabaseError as ex:
+        err_obj = ex.args
+        if err_obj[0].code == 942:
+            return False
+        raise
+
+
+def _compare_version(version: str, target_version: str) -> bool:
+    # split both version strings into parts
+    version_parts = [int(part) for part in version.split(".")]
+    target_parts = [int(part) for part in target_version.split(".")]
+
+    # compare each part
+    for v, t in zip(version_parts, target_parts):
+        if v < t:
+            return True  # current version is less
+        elif v > t:
+            return False  # current version is greater
+
+    # if all parts equal so far, check if version has fewer parts than target_version
+    return len(version_parts) < len(target_parts)
+
+
+@_handle_exceptions
+def _index_exists(connection: Connection, index_name: str) -> bool:
+    # check if the index exists
+    query = """
+        SELECT index_name 
+        FROM all_indexes 
+        WHERE upper(index_name) = upper(:idx_name)
+        """
+
+    with connection.cursor() as cursor:
+        # execute the query
+        cursor.execute(query, idx_name=index_name.upper())
+        result = cursor.fetchone()
+
+        # check if the index exists
+    return result is not None
+
+
+@_handle_exceptions
+async def _aindex_exists(connection: AsyncConnection, index_name: str) -> bool:
+    # check if the index exists
+    query = """
+        SELECT index_name 
+        FROM all_indexes 
+        WHERE upper(index_name) = upper(:idx_name)
+        """
+
+    with connection.cursor() as cursor:
+        # execute the query
+        await cursor.execute(query, idx_name=index_name.upper())
+        result = await cursor.fetchone()
+
+        # check if the index exists
+    return result is not None
+
+
+def _get_distance_function(distance_strategy: DistanceStrategy) -> str:
+    # dictionary to map distance strategies to their corresponding function
+    # names
+    distance_strategy2function = {
+        DistanceStrategy.EUCLIDEAN_DISTANCE: "EUCLIDEAN",
+        DistanceStrategy.DOT_PRODUCT: "DOT",
+        DistanceStrategy.COSINE: "COSINE",
+    }
+
+    # attempt to return the corresponding distance function
+    if distance_strategy in distance_strategy2function:
+        return distance_strategy2function[distance_strategy]
+
+    # if it's an unsupported distance strategy, raise an error
+    raise ValueError(f"Unsupported distance strategy: {distance_strategy}")
+
+
+def _get_index_name(base_name: str) -> str:
+    unique_id = str(uuid.uuid4()).replace("-", "")
+    return f"{base_name}_{unique_id}"
+
+
+def _get_table_dict(embedding_dim: int) -> Dict:
+    cols_dict = {
+        "id": "RAW(16) DEFAULT SYS_GUID() PRIMARY KEY",
+        "text": "CLOB",
+        "metadata": "JSON",
+        "embedding": f"vector({embedding_dim}, FLOAT32)",
```

Comment:
do we plan to add support for other storage types like float64/ int8,.. or flexible dimensions (*). May be we enhance the variable options later and consider them from user..